lockfile/src/golang.rs

```diff
@@ -66,4 +110,258 @@ mod tests {
             assert!(pkgs.contains(&expected_pkg));
         }
     }
+
+    #[test]
+    fn parse_go_mod() {
+        let pkgs = GoMod.parse(include_str!("../../tests/fixtures/go.mod")).unwrap();
+        assert_eq!(pkgs.len(), 5);
+
+        let expected_pkgs = [
+            Package {
+                name: "github.com/go-chi/chi/v5".into(),
+                version: PackageVersion::FirstParty("v5.0.12".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "github.com/rs/zerolog".into(),
+                version: PackageVersion::FirstParty("v1.32.0".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "github.com/mattn/go-colorable".into(),
+                version: PackageVersion::FirstParty("v0.1.13".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "github.com/mattn/go-isatty".into(),
+                version: PackageVersion::FirstParty("v0.0.20".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "golang.org/x/sys".into(),
+                version: PackageVersion::FirstParty("v0.12.0".into()),
+                package_type: PackageType::Golang,
+            },
+        ];
+
+        for expected_pkg in expected_pkgs {
+            assert!(pkgs.contains(&expected_pkg));
+        }
```

Comment:
The whole point of the `expected_pkg` stuff in our existing tests is to confirm a subset of packages are correct, while the original file itself contains a larger set of packages which possibly might cause issues.

If the test file is a minimal example specifically for testing against anyway, there's no need to check like this. You can just compare the entire vec instead. If you are concerned about order, then sorting should be pretty easy.